llvm/unittests/ADT/STLExtrasTest.cpp

```diff
@@ -532,6 +534,21 @@ TEST(STLExtrasTest, ConcatRangeADL) {
   EXPECT_THAT(concat<const int>(S0, S1), ElementsAre(1, 2, 3, 4));
 }
 
+TEST(STLExtrasTest, ConcatRangePtrToDerivedClass) {
+  auto S0 = std::make_unique<some_namespace::some_struct>();
+  auto S1 = std::make_unique<some_namespace::derives_from_some_struct>();
```

Comment:
I don't think we necessarily need unique_ptr to test this, stack allocated values would work too.

Reply:
Fixed :+1:; thanks for pointing out!